tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/multipageeditor/MultiVariablePageTest.java

```diff
@@ -155,9 +155,9 @@ public void testPostSelection() throws Throwable {
 		try {
 			postProvider.addPostSelectionChangedListener(listener);
 			editor.setPage(1);
-			assertEquals(1, fPostCalled);
-			editor.setPage(0);
```

Comment:
Is removing that necessary for the test to pass? If a test is broken, then isn't it a regression?

Reply:
i am not removing it i updated the expectation form 1 to 2, since we now get an immediate event.

Reply:
I think what's in the test is what is expected (1 event for each `setPage`), what's is the new test seems bogus: 1 `setPage` => 2 events ?

Reply:
there are two totally different senders for the both events. 1 for a page change and 1 for the included selection change which  used to come later

Reply:
ah ok, the test doesn't make it clear. Then can you please add a note about it in new comments inside the tests. Someone seeing this new test would easily be confused without such explanation.

Reply:
i now filter for the  SelectionProvider that is tested

Reply:
Perfect, thanks!